src/lib.rs

```diff
@@ -114,34 +115,34 @@ pub trait PolynomialCommitment<F: Field>: Sized {
     /// for the polynomial commitment scheme.
     fn setup(
         max_degree: usize,
-    ) -> Result<Self::UniversalParams, Self::Error>;
+    ) -> Result<Self::Parameters, Self::Error>;
 
     /// Constructs public parameters when given as input the maximum degree `degree`
     /// for the polynomial commitment scheme from given seed
     fn setup_from_seed(
         max_degree: usize,
         seed: &[u8],
-    ) -> Result<Self::UniversalParams, Self::Error>;
-
-        /// Specializes the public parameters for polynomials up to the given `supported_degree`
-    /// and for enforcing degree bounds in the range `1..=supported_degree`.
-    fn trim(
-        pp: &Self::UniversalParams,
-        supported_degree: usize,
-    ) -> Result<(Self::CommitterKey, Self::VerifierKey), Self::Error>;
+    ) -> Result<Self::Parameters, Self::Error>;
+
+    /// The low-level single segment single poly commit function.
+    /// Create a dlog commitment to `scalars` using the commitment key `comm_key`.
+    /// Optionally, randomize the commitment using `hiding_generator` and `randomizer`.
+    fn commit(
+        comm_key: &[G],
+        scalars: &[G::ScalarField],
+        hiding_generator: Option<G>,
+        randomizer: Option<G::ScalarField>,
+    ) -> Result<G::Projective, Self::Error>;
 
     /// Outputs a commitments to `polynomials`. If `polynomials[i].is_hiding()`,
-    /// then the `i`-th commitment is hiding up to `polynomials.hiding_bound()` queries.
+    /// then the `i`-th commitment is hiding
     /// `rng` should not be `None` if `polynomials[i].is_hiding() == true` for any `i`.
     ///
     /// If for some `i`, `polynomials[i].is_hiding() == false`, then the
     /// corresponding randomness is `Self::Randomness::empty()`.
-    ///
-    /// If for some `i`, `polynomials[i].degree_bound().is_some()`, then that
-    /// polynomial will have the corresponding degree bound enforced.
-    fn commit<'a>(
+    fn commit_vec<'a>(
         ck: &Self::CommitterKey,
-        polynomials: impl IntoIterator<Item = &'a LabeledPolynomial<F>>,
+        polynomials: impl IntoIterator<Item = &'a LabeledPolynomial<G::ScalarField>>,
```

Comment:
let us call the rng as in the non-vector variant.

Reply:
Done